beacon-chain/core/validators/validator.go

```diff
@@ -351,7 +359,10 @@ func UpdateRegistry(ctx context.Context, state *pb.BeaconState) (*pb.BeaconState
 				break
 			}
 			state = ExitValidator(state, uint64(idx))
-			ExitedValidators[updatedEpoch] = append(ExitedValidators[updatedEpoch], uint64(idx))
+			vStore.Lock()
```

Comment:
Does this work? I think you are still in a lock from line 326

Reply:
Ah yes. Fixed.